iceoryx_hoofs/test/moduletests/test_posix_periodic_timer.cpp

```diff
@@ -0,0 +1,222 @@
+
+TEST_F(PeriodicTimer_test, TimerAutoStartTest)
+{
+    Timer sut(0_s);
+
+    auto timerState = sut.wait(iox::posix::TimerCatchupPolicy::IMMEDIATE_TICK);
+
+    ASSERT_FALSE(timerState.has_error());
+}
+
+TEST_F(PeriodicTimer_test, ZeroINTERVALTest)
+{
+    Timer sut(0_s);
+
+    auto timerState = sut.wait(iox::posix::TimerCatchupPolicy::IMMEDIATE_TICK);
+    bool result = timerState.value().state == iox::posix::TimerState::TICK ? true : false;
+
+    ASSERT_TRUE(result);
+}
+
+TIMING_TEST_F(PeriodicTimer_test, DurationINTERVALTest, Repeat(5), [&] {
+    Timer sut(INTERVAL);
+
+    auto timeBeforeWait = sut.now().value();
+    auto timerState = sut.wait(iox::posix::TimerCatchupPolicy::IMMEDIATE_TICK);
+    auto timeAfterWait = sut.now().value();
+    auto duration = timeAfterWait - timeBeforeWait;
+    bool result = (duration.toMilliseconds() == INTERVAL.toMilliseconds()) ? true : false;
+
+    TIMING_TEST_EXPECT_FALSE(timerState.has_error());
+    TIMING_TEST_EXPECT_TRUE(result);
+});
+
+TEST_F(PeriodicTimer_test, TimerStopTest)
+{
+    Timer sut(0_s);
+
+    sut.stop();
+    auto timerState = sut.wait(iox::posix::TimerCatchupPolicy::IMMEDIATE_TICK);
+    bool result = timerState.value().state == iox::posix::TimerState::STOP ? true : false;
+
+    ASSERT_TRUE(result);
+}
+
+
+TEST_F(PeriodicTimer_test, TimerStopAfterWaitTest)
+{
+    Timer sut(INTERVAL);
+
+    auto timerState = sut.wait(iox::posix::TimerCatchupPolicy::IMMEDIATE_TICK);
+    sut.stop();
+    timerState = sut.wait(iox::posix::TimerCatchupPolicy::IMMEDIATE_TICK);
+    bool result = timerState.value().state == iox::posix::TimerState::STOP ? true : false;
+
+    ASSERT_TRUE(result);
+}
+
+
+TIMING_TEST_F(PeriodicTimer_test, ResetWithNewDurationINTERVALTest, Repeat(5), [&] {
+    Timer sut(INTERVAL);
+    iox::units::Duration NEW_DURATION{100_ms};
+    const int RANGE_APPROX = 2; // 2ms arppoximation. This may be lost in execution.
+    sut.start(NEW_DURATION);
+
+    auto timeBeforeWait = sut.now().value();
+    auto timerState = sut.wait(iox::posix::TimerCatchupPolicy::IMMEDIATE_TICK);
+    auto timeAfterWait = sut.now().value();
+    auto duration = timeAfterWait - timeBeforeWait;
+
+    TIMING_TEST_EXPECT_FALSE(timerState.has_error());
+    TIMING_TEST_EXPECT_TRUE(duration.toMilliseconds() - NEW_DURATION.toMilliseconds() <= RANGE_APPROX);
+});
+
+TIMING_TEST_F(PeriodicTimer_test, currentTimeTest, Repeat(5), [&] {
+    Timer sut(INTERVAL);
+    struct timespec ts;
+
+    clock_gettime(CLOCK_REALTIME, &ts);
+    auto timeNow = sut.now().value();
+    auto currentSystemTime = iox::units::Duration(ts);
+
+    bool result = timeNow.toMilliseconds() == currentSystemTime.toMilliseconds() ? true : false;
+
+    TIMING_TEST_EXPECT_TRUE(result);
+});
+
+TIMING_TEST_F(PeriodicTimer_test, periodicityWithoutExecutionTimeTest, Repeat(5), [&] {
+    Timer sut(INTERVAL);
+    const int RANGE_APPROX = 2; // 2ms arppoximation. This may be lost in execution.
```

Comment:
Nitpick
```suggestion
    const int RANGE_APPROX = 2; // 2ms approximation. This may be lost in execution.
```

Reply:
:) Thanks. This typo is corrected. Since its also present in 3 other locations I didnt apply the commit suggestion.